### ISSUE TITLE
Multiplying `npt.NDArray[np.float64]` with `npt.NDArray[np.float64]` or `float` outputs `npt.NDArray[np.double]`

### ISSUE THREAD

**Issue by Saransh-cpp** [python/mypy#18561]
I am not sure if this should be reported in NumPy or mypy but my projects started seeing this after mypy's recent release; hence, reporting it here. Please let me know if I should move this!

**Bug Report**

Mypy thinks multiplying `npt.NDArray[np.float64]` with a `float` or another `npt.NDArray[np.float64]` should result in `npt.NDArray[np.double]`, and I think that should not be the case. Before the latest release, having `npt.NDArray[np.float64]` as the return type would satisfy mypy but now it fails. 

**To Reproduce**

```python
import numpy.typing as npt
import numpy as np


def do_something(a: npt.NDArray[np.float64], b: npt.NDArray[np.float64]) -> npt.NDArray[np.float64]:
    return a * b
```

**Expected Behavior**

Running `mypy` should pass.

**Actual Behavior**

```py
mypy_rep.py:6: error: Incompatible return value type (got "ndarray[tuple[int, ...], dtype[floating[Any]]]", expected "ndarray[tuple[int, ...], dtype[float64]]")  [return-value]
Found 1 error in 1 file (checked 1 source file)
```

**Your Environment**

- Mypy version used: 1.14.1 (compiled: yes)
- Mypy command-line flags: none
- Mypy configuration options from `mypy.ini` (and other config files): none
- Python version used: 3.13.1

**Comment by A5rocks** [comment]
Interesting, I would assume this overload would get selected:

```py
    @overload
    def __mul__(self: _ArrayFloat64_co, other: _ArrayLike[floating[_64Bit]], /) -> NDArray[float64]: ...
```

Maybe a `git bisect` would help clarify if this is a mypy or numpy bug?

**Comment by A5rocks** [comment]
I'm unable to reproduce this on mypy v1.14.1 with numpy 2.2.2, so I'll close this. Feel free to reopen (I think you can? or just tell me to reopen this) if I was just missing something.

**Comment by Saransh-cpp** [comment]
Ah, I was getting this on `numpy-2.2.0`. Thanks for looking into this; upgrading numpy did the work!
